clients/imodels-client-management/src/base/internal/OperationsBase.ts

```diff
@@ -97,7 +98,28 @@ export class OperationsBase<TOptions extends OperationsBaseOptions> {
     };
   }
 
-  private async formHeaders(params: AuthorizationParam & { preferReturn?: PreferReturn, contentType?: ContentType }): Promise<Dictionary<string>> {
+  private resolveHeaderValue(headerOrHeaderFactory: (() => string | undefined) | string): string | undefined {
+    if (typeof headerOrHeaderFactory === "function")
+      return headerOrHeaderFactory();
+    return headerOrHeaderFactory;
+  }
+
+  private addOrUpdateHeaders(existingHeaders: Dictionary<string>, additionalHeaders?: HeaderFactories) {
+    if (!additionalHeaders)
+      return;
+
+    for (const headerName in additionalHeaders) {
+      if(Object.prototype.hasOwnProperty.call(additionalHeaders, headerName)) {
+        const headerValue: string | undefined = this.resolveHeaderValue(additionalHeaders[headerName]);
+        if (typeof headerValue === "string")
```

Comment:
what if header value is ""? or "  "? would be nice to add a note about this behavior to the documentation section you've written.

in terms of documentation, it would ne nice to mention here that param headers will overwrite constructor headers - 
![image](https://github.com/iTwin/imodels-clients/assets/70565417/cd7332d8-fe4d-4420-9532-4eb0f790c789)
This comment is shown when developers write code so it has pretty good visibility. It is sufficient to mention edge cases in the README docs but the most important case about overwriting could be mentioned in the param documentation.